### ISSUE TITLE
image archive use case: limiting CPU usage of computational intensive operations

### ISSUE THREAD

**Issue by kswang1029** [CARTAvis/carta-node-server#32]
NOTE: please move this issue to a proper repo if this repo is not the right place

For an image archive such the ALMA science archive, the main use case is to view images quickly instead of performing analysis directly using CARTA as a platform. Thus in this case, the archive would like to limit the CPU usage of some intensive calculations while keeping the performance of image "viewing" functionalities. In this case, operations include:
- cube histogram (multi-threaded)
- moment map generation (this is currently single-threaded. no need to limit it. need to revisit this once parallelized)
- PV map generation (this will be potentially single-threaded. no need to limit it. need to revisit this if parallelized)
- animation playback (multi-threaded)

**Comment by veggiesaurus** [comment]
Animation playback is single-threaded.

It wasn't entirely clear to me whether you just wanted to limit the number of threads (this is already possible if you use the `OMP_NUM_THREADS` environmental variable) or actually disable this functionality entirely (that would be more of a frontend and backend issue)

**Comment by kswang1029** [comment]
When animation playback is on, I see my CPU cores are fully occupied. Due to tile generation?

**Comment by kswang1029** [comment]
> Animation playback is single-threaded.
> 
> It wasn't entirely clear to me whether you just wanted to limit the number of threads (this is already possible if you use the `OMP_NUM_THREADS` environmental variable) or actually disable this functionality entirely (that would be more of a frontend and backend issue)

The idea here is to limit only those not-that-essential tools for the ALMA archive. If we use OMP_NUM_THREADS to globally limit the CPU usage, viewing images will be slowing down too.

**Comment by veggiesaurus** [comment]
> When animation playback is on, I see my CPU cores are fully occupied. Due to tile generation?

yes. So this is true for image viewing in general as well. I guess we could limit the FPS or something?

**Comment by kswang1029** [comment]
@kswang1029 to refine the requirements
